src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1179,6 +1179,22 @@ def filter(self, col: Expression) -> "SelectStatement":
                 from_=self.to_subqueryable(), where=col, analyzer=self.analyzer
             )
 
+        # extract any subquery plan from ScalarSubquery expression that may be
+        # present in the col expression.
+        self.analyzer.subquery_plans = []
+        self.analyzer.analyze(col, self.df_aliased_col_name_to_real_col_name)
```

Comment:
this relies on the fact that analyzer.subquery_plans is empty at that time, but this may not be true.
I think the right way to handle this is probably here https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L778, after the resolve, we need to propogate the subquery correctly. Right now, we only handle this for snowlake plan, however, after sql simplification is applied, both SelectStatement and SnowflakePlan are resolved plan, so we should also propagate to the source plan if the source plan is a selectstatement to make sure all nodes are resolved properly